darks_driftwind_crypt/developmental_testing.dsc

```diff
@@ -0,0 +1,166 @@
+#13 I also need a system that will teleport the player to the start of the room if they step on an incorrect pressure plate
+  #- The solution is on the ceiling in wool
+
+    after player right clicks lectern:
+      - if <player.has_flag[region_six.has_book]>:
+        - stop
+      - flag player region_six.has_book
+      - give <context.location.inventory.list_contents.first>
+    after player exits region_six:
+      - if <player.inventory.contains_item[written_book]>:
+        - take item:written_book
+
+    # ! bad pressure plates would just be easily defined by flagging them via `/ex flag <player.cursor_on> bad_pressure_plate`
+    after player stands on *_pressure_plate:
+      - if <context.location.has_flag[bad_pressure_plate]>
+      - run fancy_painful_teleport defmap:teleport=region_six_teleport_respawn
+
+# @ #14 I need this door to open when the player has finished the puzzle
+    after player enters region_eight_name:
+      - run open_gate_task defmap:door=three
+
+# @ #15 I need a transition to the large hallway to the left of this block
+    after player enters region_nine_name:
+      - run fancy_teleport defmap:teleport=room_four_name
+
+#17 Npc named "Vel", what he says is tbd, need him to vanish when he finishes talking as if he died
+    # @ see below maybe
+    on npc death:
+      - if <npc.script.name.if_null[invalid]> == vel:
+        - determine "well, i'm pretty sure he was going to say Behr was the coolest ever but he died a very spontaneous and what seemed to be quick death."
+
+#18 Door that needs to be opened *after* the npc vanishes
+      - wait 3s
+      - run open_gate_task defmap:door=four
+
+#19 - No block - The entire hall is where the boss fight will take place
+
+# misc:
+    after reload scripts:
+      - define generic_location <location[0,0,0,<server.worlds.first.name>]>
+      - foreach region_one_name|region_two_name|yellow_buttons|red_buttons|blue_buttons|gray_buttons|region_four_name|region_five_name|region_six_name|region_seven_name|region_eight_name|region_nine_name as:area:
+        - if !<cuboid[<[area]>].exists>:
+          - note <[generic_location].to_cuboid[<[generic_location]>]> as:<[area]>
+
+fancy_teleport:
+  type: task
+  definitions: location
+  script:
+  # TODO: This title insertion / fade depends on: - resource pack data and structure
+    - title title:<&font[fade:black]><&chr[0004]><&chr[F801]><&chr[0004]> fade_in:5t stay:0s fade_out:1s
+    - wait 5t
+    - teleport <[location]>
+
+fancy_painful_teleport:
+  type: task
+  definitions: location
+  script:
+  # TODO: This title insertion / fade depends on: - resource pack data and structure
+    - animate <player> animation:hurt_drown
+    - title title:<&font[fade:black]><&chr[0004]><&chr[F801]><&chr[0004]> fade_in:2t stay:0s fade_out:1s
+    - wait 2t
+    - teleport <[location]>
```

Comment:
these specifically utilize my own resource pack as mentioned in the description of the pull request; the method utilizing `<&font[fade:black]><&chr[0004]>` calls my own resource pack that i have to verify the script functions correctly - this will need to be changed to what the active pack defines screen overlays / "fades" to create a transitional effect